solutions/Assignment2_1.java

```
@@ -0,0 +1,206 @@
+/*JAVA program to implement Multiple client-server interaction
+ *Multithreading approach is used where Server and Clients run as separate threads
+ *Server has an additional ClientServiceThread which processes the requests of the clients
+ *The Server accepts requests from Clients and passes the requests to ClientServiceThread
+ *All messages are recorded in a LOG File
+ */
+import java.io.*;
+import java.net.*;
+import java.util.*;
+import java.util.logging.*;
+
+// Server class
+class Server extends Thread {
+  private static final Logger logger = Logger.getLogger(Server.class.getName());
+  static int port;
+  static ServerSocket serverSocket;
+  static Socket socket;
+  // Constructor
+  public Server(int port) throws Exception {
+      Server.port = port;
+      serverSocket = new ServerSocket(port);
+    }
+  @Override
+  public void run() {
+    logger.log(Level.INFO, "Server Started");
+    while (true) {
+      logger.log(Level.INFO, "Waiting for client to connect");
+      try {
+        socket = serverSocket.accept();
+      } catch(Exception e) {}
+      logger.log(Level.INFO, "Client connected");
+      ClientServiceThread worker = new ClientServiceThread(socket);
+      worker.start();
+    }
+  }
+}
+
+// Class which is responsible for processing Client request
+class ClientServiceThread extends Thread {
+  Socket socket;
+  //Constructor
+  public ClientServiceThread(Socket socket) {
+    this.socket = socket;
+  }
+  @Override
+  public void run() {
+    try {
+      Thread.sleep(2000); // Wait for 2 seconds
+    } catch(Exception e) {}
```

In general, catch-all exceptions are a bad idea, since they hide information that could otherwise be useful.
In fact, I'd rather not have exceptions caught at all, unless it is information that you actively use.
What's a valid usecase of catching? Database connection failure, since that is something you can retry.
In most cases, just let it bubble up to `main`, which can then figure out what to do.